app installation does not support paths with spaces

**Describe the bug**
Have a JBang script in a folder with a space in it. Run `jbang app install` and try to run the installed app

```
[jbang] [ERROR] Script or alias could not be found or read: 'C:\work\src\github.com\jeffmaury\kuben'
[jbang] Run with --verbose for more details
```

**To Reproduce**
Steps to reproduce the behavior:
1. Create a JBang script with space in folder
2. run j`bang app install ...`
3. run installed app
4. See error

**Expected behavior**
App should run

**JBang version**
```
[jbang] jbang version 0.98.0
Cache: C:\Users\JeffMAURY\.jbang\cache
Config: C:\Users\JeffMAURY\.jbang
Repository:C:\Users\JeffMAURY\.m2\repository
0.98.0
```

**Additional context**
Add any other context about the problem here.

--- a/itests/with space/helloworld.java
+++ b/itests/with space/helloworld.java
@@ -1,0 +1,13 @@
+///usr/bin/env jbang "$0" "$@" ; exit $?
+// //DEPS <dependency1> <dependency2>
+//JAVA_OPTIONS -Dfoo=bar "-Dbar=aap noot mies"
+//MANIFEST foo bar=baz baz=${bazprop:nada}
+
+import static java.lang.System.*;
+
+public class helloworld {
+
+    public static void main(String... args) {
+        out.println("Hello " + (args.length>0?args[0]:"World"));
+    }
+}

--- a/src/main/java/dev/jbang/cli/App.java
+++ b/src/main/java/dev/jbang/cli/App.java
@@ -27,6 +27,7 @@ import dev.jbang.dependencies.DependencyUtil;
 import dev.jbang.net.JdkManager;
 import dev.jbang.source.Project;
 import dev.jbang.source.RunContext;
+import dev.jbang.util.CommandBuffer;
 import dev.jbang.util.UnpackUtil;
 import dev.jbang.util.Util;
 
@@ -140,9 +141,13 @@ class AppInstall extends BaseCommand {
 	}
 
 	private static void installShellScript(Path file, String scriptRef, boolean benative) throws IOException {
-		List<String> lines = Arrays.asList(
-				"#!/bin/sh",
-				"exec jbang run" + (benative ? " --native " : " ") + scriptRef + " \"$@\"");
+		CommandBuffer cb;
+		if (benative) {
+			cb = CommandBuffer.of("exec", "jbang", "run", "--native", scriptRef);
+		} else {
+			cb = CommandBuffer.of("exec", "jbang", "run", scriptRef);
+		}
+		List<String> lines = Arrays.asList("#!/bin/sh", cb.asCommandLine(Util.Shell.bash) + " \"$@\"");
 		Files.write(file, lines, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.CREATE);
 		if (!Util.isWindows()) {
 			setExecutable(file);
@@ -162,15 +167,24 @@ class AppInstall extends BaseCommand {
 	}
 
 	private static void installCmdScript(Path file, String scriptRef, boolean benative) throws IOException {
-		List<String> lines = Arrays.asList(
-				"@echo off",
-				"jbang run" + (benative ? " --native " : " ") + scriptRef + " %*");
+		CommandBuffer cb;
+		if (benative) {
+			cb = CommandBuffer.of("jbang", "run", "--native", scriptRef);
+		} else {
+			cb = CommandBuffer.of("jbang", "run", scriptRef);
+		}
+		List<String> lines = Arrays.asList("@echo off", cb.asCommandLine(Util.Shell.cmd) + " %*");
 		Files.write(file, lines, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.CREATE);
 	}
 
 	private static void installPSScript(Path file, String scriptRef, boolean benative) throws IOException {
-		List<String> lines = Collections.singletonList(
-				"jbang run" + (benative ? " --native " : " ") + scriptRef + " @args");
+		CommandBuffer cb;
+		if (benative) {
+			cb = CommandBuffer.of("jbang", "run", "--native", scriptRef);
+		} else {
+			cb = CommandBuffer.of("jbang", "run", scriptRef);
+		}
+		List<String> lines = Collections.singletonList(cb.asCommandLine(Util.Shell.powershell) + " @args");
 		Files.write(file, lines, StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.CREATE);
 	}
 

--- a/src/test/java/dev/jbang/cli/TestApp.java
+++ b/src/test/java/dev/jbang/cli/TestApp.java
@@ -24,18 +24,18 @@ import dev.jbang.util.Util;
 
 public class TestApp extends BaseTest {
 	private static final List<String> shContents = Arrays.asList("#!/bin/sh",
-			"exec jbang run $CWD/itests/helloworld.java \"$@\"");
+			"exec jbang run '$CWD/itests/with space/helloworld.java' \"$@\"");
 	private static final List<String> cmdContents = Arrays.asList("@echo off",
-			"jbang run $CWD\\itests\\helloworld.java %*");
+			"jbang run ^\"$CWD\\itests\\with^ space\\helloworld.java^\" %*");
 	private static final List<String> ps1Contents = Collections.singletonList(
-			"jbang run $CWD\\itests\\helloworld.java @args");
+			"jbang run '$CWD\\itests\\with space\\helloworld.java' @args");
 
 	private static final List<String> nativeShContents = Arrays.asList("#!/bin/sh",
-			"exec jbang run --native $CWD/itests/helloworld.java \"$@\"");
+			"exec jbang run --native '$CWD/itests/with space/helloworld.java' \"$@\"");
 	private static final List<String> nativeCmdContents = Arrays.asList("@echo off",
-			"jbang run --native $CWD\\itests\\helloworld.java %*");
+			"jbang run --native ^\"$CWD\\itests\\with^ space\\helloworld.java^\" %*");
 	private static final List<String> nativePs1Contents = Collections.singletonList(
-			"jbang run --native $CWD\\itests\\helloworld.java @args");
+			"jbang run --native '$CWD\\itests\\with space\\helloworld.java' @args");
 
 	private static final List<String> h2shContents = Arrays.asList("#!/bin/sh",
 			"exec jbang run com.h2database:h2:1.4.200 \"$@\"");
@@ -46,7 +46,7 @@ public class TestApp extends BaseTest {
 
 	@Test
 	void testAppInstallFile() throws IOException {
-		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String src = examplesTestFolder.resolve("with space/helloworld.java").toString();
 		ExecutionResult result = checkedRun(null, "app", "install", src);
 		assertThat(result.err, containsString("Command installed: helloworld"));
 		if (Util.isWindows()) {
@@ -59,7 +59,7 @@ public class TestApp extends BaseTest {
 
 	@Test
 	void testAppNativeInstallFile() throws IOException {
-		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String src = examplesTestFolder.resolve("with space/helloworld.java").toString();
 		ExecutionResult result = checkedRun(null, "app", "install", "--native", src);
 		assertThat(result.err, containsString("Command installed: helloworld"));
 		if (Util.isWindows()) {
@@ -119,7 +119,7 @@ public class TestApp extends BaseTest {
 
 	@Test
 	void testAppInstallFileExists() throws IOException {
-		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String src = examplesTestFolder.resolve("with space/helloworld.java").toString();
 		ExecutionResult result = checkedRun(null, "app", "install", src);
 		assertThat(result.err, containsString("Command installed: helloworld"));
 		if (Util.isWindows()) {
@@ -134,7 +134,7 @@ public class TestApp extends BaseTest {
 
 	@Test
 	void testAppInstallFileForce() throws IOException {
-		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String src = examplesTestFolder.resolve("with space/helloworld.java").toString();
 		ExecutionResult result = checkedRun(null, "app", "install", src);
 		assertThat(result.err, containsString("Command installed: helloworld"));
 		if (Util.isWindows()) {
@@ -180,7 +180,7 @@ public class TestApp extends BaseTest {
 
 	@Test
 	void testAppInstallFileWithName() throws IOException {
-		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String src = examplesTestFolder.resolve("with space/helloworld.java").toString();
 		ExecutionResult result = checkedRun(null, "app", "install", "--name=hello", src);
 		assertThat(result.err, containsString("Command installed: hello"));
 		if (Util.isWindows()) {
@@ -195,7 +195,7 @@ public class TestApp extends BaseTest {
 
 	@Test
 	void testAppInstallFileWithNameExists() throws IOException {
-		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String src = examplesTestFolder.resolve("with space/helloworld.java").toString();
 		ExecutionResult result = checkedRun(null, "app", "install", "--name=hello", src);
 		assertThat(result.err, containsString("Command installed: hello"));
 		if (Util.isWindows()) {
@@ -210,7 +210,7 @@ public class TestApp extends BaseTest {
 
 	@Test
 	void testAppInstallFileWithNameForce() throws IOException {
-		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String src = examplesTestFolder.resolve("with space/helloworld.java").toString();
 		ExecutionResult result = checkedRun(null, "app", "install", "--name=hello", src);
 		assertThat(result.err, containsString("Command installed: hello"));
 		if (Util.isWindows()) {
@@ -224,7 +224,7 @@ public class TestApp extends BaseTest {
 
 	@Test
 	void testAppInstallAlias() throws IOException {
-		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String src = examplesTestFolder.resolve("with space/helloworld.java").toString();
 		checkedRun(null, "alias", "add", "-g", "--name=apptest", src);
 		ExecutionResult result = checkedRun(null, "app", "install", "apptest");
 		assertThat(result.err, containsString("Command installed: apptest"));
@@ -240,7 +240,7 @@ public class TestApp extends BaseTest {
 
 	@Test
 	void testAppInstallAliasFromRepo() throws IOException {
-		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String src = examplesTestFolder.resolve("with space/helloworld.java").toString();
 		checkedRun(null, "alias", "add", "-g", "--name=apptest", src);
 		checkedRun(null, "catalog", "add", "-g", "--name=testrepo",
 				jbangTempDir.resolve("jbang-catalog.json").toString());
@@ -279,7 +279,7 @@ public class TestApp extends BaseTest {
 
 	@Test
 	void testAppList() throws IOException {
-		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String src = examplesTestFolder.resolve("with space/helloworld.java").toString();
 		checkedRun(null, "app", "install", "--name=hello1", src);
 		checkedRun(null, "app", "install", "--name=hello2", src);
 		checkedRun(null, "app", "install", "--name=hello3", src);
@@ -290,7 +290,7 @@ public class TestApp extends BaseTest {
 
 	@Test
 	void testAppUninstall() throws IOException {
-		String src = examplesTestFolder.resolve("helloworld.java").toString();
+		String src = examplesTestFolder.resolve("with space/helloworld.java").toString();
 		checkedRun(null, "app", "install", src);
 		if (Util.isWindows()) {
 			assertThat(Settings.getConfigBinDir().resolve("helloworld.cmd").toFile(), anExistingFile());

